src/components/general/Dropdown/index.tsx

```diff
@@ -0,0 +1,143 @@
+    const [inputValue, setInputValue] = useState('');
+    const [dropdownSelections, setDropdownSelections] = useState(selections);
+
+    const inputRef = useRef<HTMLInputElement | null>(null);
+    const inputContainerRef = useRef<HTMLDivElement | null>(null);
+
+    const close = useCallback(() => open && setOpen(false), [open]);
+
+    const filterSearch = useCallback(inputValue => {
+        setDropdownSelections(
+            selections.filter(selection =>
+                selection.title.toLowerCase().includes(inputValue.toLowerCase())
+            )
+        );
+    }, []);
+
+    useClickOutsideOverlayPortal(close, inputContainerRef.current);
+
+    return (
+        <div className={styles.inputContainer} ref={inputContainerRef}>
+            <TextInput
+                onChange={value => {
+                    if (!open) {
+                        setOpen(true);
+                        return;
+                    }
+                    setInputValue(value);
+                    filterSearch(value);
+                }}
+                label={label}
+                icon={<DropdownArrow cursor="pointer" isOpen={open} />}
+                onIconAction={() => open && setOpen(false)}
+                onClick={() => !open && setOpen(true)}
+                value={open ? inputValue : selected}
+                ref={inputRef}
+            />
+            <Overlay relativeRef={inputContainerRef} show={open}>
+                <div className={styles.menuContainer}>
+                    <Menu
+                        onClick={item => {
+                            console.log(open);
+                            if (open) {
+                                onSelect && onSelect(item);
+                                setOpen(false);
+                            }
+                        }}
+                        keyboardNavigationRef={inputRef.current}
+                        sections={[
+                            {
+                                key: 'DropdownSelection',
+                                items: dropdownSelections,
+                            },
+                        ]}
+                    />
+                </div>
+            </Overlay>
+        </div>
+    );
+};
+
+type OverlayProps = {
+    children?: ReactNode,
+    relativeRef: MutableRefObject<HTMLElement | null>,
+    show?: boolean
+
+}
+
+const Overlay: FC<OverlayProps> = ({ children, relativeRef, show }) => {
+    const fusionContext = useFusionContext();
+    const element = document.createElement('div');
+    const overlayContainer = useOverlayContainer();
+    const rect = useRelativePositioning(relativeRef);
+
+    useEffect(() => {
+        overlayContainer && overlayContainer.appendChild(element);
+
+        return () => {
+            overlayContainer && overlayContainer.removeChild(element);
+        };
+    }, []);
+    if(show===false) {
+        return null;
+    }
+    return (
+        overlayContainer &&
```

Comment:
Could combine this with if(show === false) to group the conditions for when nothing should be rendered. Easier for other developers to read and understand  the code